File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/CustomerDao.java
package by.bstu.vs.stpms.courier_application.model.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

import by.bstu.vs.stpms.courier_application.model.database.entity.Customer;
import by.bstu.vs.stpms.courier_application.model.database.entity.Product;

@Dao
public abstract class CustomerDao extends AbstractDao<Customer> {
    @Override
    @Query("SELECT * FROM customer WHERE id = :id")
    public abstract LiveData<Customer> findById(long id);
    @Override
    @Query("SELECT * FROM customer")
    public abstract LiveData<List<Customer>> getAll();
}

File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/CustomerDao.kt
package by.bstu.vs.stpms.courier_application.model.database.dao

import androidx.room.*
import by.bstu.vs.stpms.courier_application.model.database.entity.Customer

@Dao
abstract class CustomerDao : AbstractDao<Customer>() {

    @Query("DELETE FROM customer WHERE 1=1")
    abstract override suspend fun truncate()

    @Query("SELECT * FROM customer WHERE id = :id")
    abstract override suspend fun findById(id: Long): Customer?

    @Query("SELECT * FROM customer")
    abstract override suspend fun getAll(): List<Customer>
}